frb_codegen/src/generator/dart/mod.rs

```diff
@@ -467,53 +620,106 @@ fn generate_api_func(func: &IrFunc, ir_file: &IrFile) -> GeneratedApiFunc {
     }
 }
 
-fn generate_api2wire_func(ty: &IrType, ir_file: &IrFile, block_index: BlockIndex) -> String {
-    if let Some(body) = TypeDartGenerator::new(ty.clone(), ir_file, None).api2wire_body(block_index)
-    {
+fn rust_wire_to_dart_wire(ty: &str) -> Cow<str> {
+    let ty = ty.trim();
+    if matches!(ty, "void" | "WireSyncReturnStruct") {
+        return ty.into();
+    }
+    if ty.starts_with("*mut") {
+        return format!("int /* {} */", ty).into();
+    }
+    return format!("dynamic /* {} */", ty).into();
+}
+
+fn generate_wasm_wire_func_decl(func: &IrFuncLike) -> String {
+    format!(
+        "external {} {name}({});",
+        rust_wire_to_dart_wire(&func.output),
+        func.inputs
+            .iter()
+            .map(|(key, ty)| format!("{} {}", ty, key))
+            .join(","),
+        name = func.name,
+    )
+}
+
+fn generate_wasm_wire_func_method(func: &IrFuncLike) -> String {
+    format!(
+        "{out} {name}({}) => wasmModule.{name}({});",
+        func.inputs
+            .iter()
+            .map(|(key, ty)| format!("{} {}", ty, key))
+            .join(","),
+        func.inputs.iter().map(|(key, _)| key.to_string()).join(","),
+        name = func.name,
+        out = if func.has_port_argument {
+            "void".into()
+        } else {
+            rust_wire_to_dart_wire(&func.output)
+        },
+    )
+}
+
+fn generate_api2wire_func(ty: &IrType, ir_file: &IrFile, config: &Opts) -> Acc<String> {
+    let Acc { common, io, wasm } =
+        TypeDartGenerator::new(ty.clone(), ir_file, None, config).api2wire_body();
+    let wrapper = |body: &str, wasm: bool| {
         format!(
-            "{} _api2wire_{}({} raw) {{
-            {}
-        }}
-        ",
-            ty.dart_wire_type(),
+            "@protected
+            {} api2wire_{}({} raw) {{
+                {}
+            }}",
+            ty.dart_wire_type(wasm),
```

Comment:
Looks like `dart_wire_type` has `wasm: bool` as function arg. Maybe we should use enum instead: `enum SomeEnumName {wasm, io}`. Then it is impossible to confuse the reader.

(Not sure whether have provided the suggestion - the PR is so large that I have forgotten what I have reviewed before :) )